tests/test_plan.py

```diff
@@ -43,6 +44,30 @@ def _final_kwargs_spec(plan):
     return plan[-1][1]
 
 
+@pytest.fixture
+def captured_errors(monkeypatch):
+    """
+    Return a dict that will capture the last errors raised by ``andi.plan``
+    """
```

Comment:
I wonder if it makes more sense to store all the structure of the errors in an attribute of NotProvidableError instead.